### PR TITLE
Complete V2 API

### LINKED ISSUE #78: Flesh out V2 API

### PR DESCRIPTION
**Pull request by harry1453**
I am aware that this is missing unit tests - these will come at a later date.

Once this is done the `brsApi.proto` file can be used to start building proper clients.

Some of the proto definitions are incompatible with 2.3.0 - these, however, were unused in 2.3.0 so the incompatibility is OK. The mining definitions (the only ones implemented in 2.3.0) have not changed

Closes #78

### CODE REVIEW

File: src/brs/grpc/proto/BrsService.java [hunk 1]
```diff
@@ -17,90 +21,295 @@
...
     }
 
     @Override
     public void getMiningInfo(Empty request, StreamObserver<BrsApi.MiningInfo> responseObserver) {
-        try {
-            getHandler(GetMiningInfoHandler.class).handleRequest(request, responseObserver);
-        } catch (HandlerNotFoundException e) {
-            responseObserver.onError(ProtoBuilder.buildError(e));
-        }
+        handleRequest(GetMiningInfoHandler.class, request, responseObserver);
```
Comment (ohager) [inline src/brs/grpc/proto/BrsService.java]: :clap:

File: src/brs/grpc/proto/ProtoBuilder.java [hunk 2]
```diff
@@ -40,12 +44,13 @@ public static StatusException buildError(Throwable t) {
         return builder.build();
     }
 
-    private static BrsApi.AssetBalance buildAssetBalance(Account.AccountAsset asset) {
+    public static BrsApi.AssetBalance buildAssetBalance(Account.AccountAsset asset) {
```
Comment (ohager) [inline src/brs/grpc/proto/ProtoBuilder.java]: Kudos for Builder Pattern

File: src/brs/http/GetConstants.java [hunk 3]
```diff
@@ -14,169 +15,67 @@
...
-    peerState.addProperty("value", 1);
-    peerState.addProperty("description", "Connected");
-    peerStates.add(peerState);
-    peerState = new JsonObject();
-    peerState.addProperty("value", 2);
-    peerState.addProperty("description", "Disconnected");
-    peerStates.add(peerState);
-    response.add("peerStates", peerStates);
+        JsonObject requestTypes = new JsonObject();
+        // for (Map.Entry<String, APIServlet.APIRequestHandler> handlerEntry : APIServlet.apiRequestHandlers.entrySet()) {
```
Comment (ohager) [inline src/brs/http/GetConstants.java]: commented code?

Comment (harry1453) [reply src/brs/http/GetConstants.java]: Yeah needs to be removed but outside of scope of this PR, there are many many instances of commented code like this

File: src/brs/grpc/proto/BrsService.java (hunk 1 above)
Comment (harry1453) [reply src/brs/grpc/proto/BrsService.java]: Took some confusing generics but got it figured out.

**Pull request merged**